mockgcp/mockcompute/networksv1.go

```diff
@@ -127,3 +129,34 @@ func (s *NetworksV1) Delete(ctx context.Context, req *pb.DeleteNetworkRequest) (
 
 	return s.newLRO(ctx, name.Project.ID)
 }
+
+type networkName struct {
+	Project *projects.ProjectData
+	Name    string
+}
+
+func (n *networkName) String() string {
+	return "projects/" + n.Project.ID + "/global" + "/networks/" + n.Name
+}
+
+// parseNetworkName parses a string into a networkName.
+// The expected form is `projects/*/global/networks/*`.
+func (s *MockService) parseNetworkName(name string) (*networkName, error) {
+	tokens := strings.Split(name, "/")
+
+	if len(tokens) == 5 && tokens[0] == "projects" && tokens[2] == "global" && tokens[3] == "networks" {
+		project, err := s.projects.GetProjectByID(tokens[1])
+		if err != nil {
+			return nil, err
+		}
+
+		name := &networkName{
+			Project: project,
+			Name:    tokens[4],
+		}
+
+		return name, nil
+	} else {
+		return nil, status.Errorf(codes.InvalidArgument, "name %q is not valid", name)
```

Comment:
Can we add either the token length or expected name format to the error message?

Reply:
I'm not sure it gets us anything?  We would not expect this error path to be hit, because we wouldn't expect terraform or the go SDK to construct invalid URLs - and even if they did, to reach this function the URLs have to be valid enough to get through the `google.api.http` grpc-gateway matching, e.g. here the proto matching is this:

```
  // Creates a network in the specified project using the data included in the request.
  rpc Insert(InsertNetworkRequest) returns (Operation) {
    option (google.api.http) = {
      body: "network_resource"
      post: "/compute/v1/projects/{project}/global/networks"
    };
    option (google.api.method_signature) = "project,network_resource";
    option (mockgcp.cloud.operation_service) = "GlobalOperations";
  }
```

So it has to be of the form `/compute/v1/projects/{project}/global/networks` anyway

And compute is actually the least strict here, most other protos look more like this:

```
  // Creates a new Network in a given project and location.
  rpc CreateNetwork(CreateNetworkRequest)
      returns (google.longrunning.Operation) {
    option (google.api.http) = {
      post: "/v1/{parent=projects/*/locations/*/zones/*}/networks"
      body: "network"
    };
    option (google.api.method_signature) = "parent,network,network_id";
    option (google.longrunning.operation_info) = {
      response_type: "Network"
      metadata_type: "OperationMetadata"
    };
  }
```